src/SFML/System/Clock.cpp

```diff
@@ -31,37 +31,102 @@
 
 namespace sf
 {
+
+////////////////////////////////////////////////////////////
+Clock::Clock() noexcept :
+m_refPoint(ClockImpl::now()),
+m_pausePoint()
+{
+
+}
+
+
 ////////////////////////////////////////////////////////////
-Clock::Clock() :
-m_startTime(ClockImpl::now())
+bool Clock::hasTimePassed(const Time& time) const noexcept
 {
+    return time.asMicroseconds() < getElapsedTime().asMicroseconds();
 }
 
 
 ////////////////////////////////////////////////////////////
-Time Clock::getElapsedTime() const
+Time Clock::getElapsedTime() const noexcept
 {
-    return durationToTime(ClockImpl::now() - m_startTime);
+    if (isRunning())
+    {
+        return sf::microseconds((ClockImpl::now() - m_refPoint).count());
```

Comment:
This implicitly requires `ClockImpl` to give time in microseconds, which might not be the case.

Reply:
Addressed in new commits.